File: solutions/python3/0101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        return self.helper(root, root)

    def helper(self, p: TreeNode, q: TreeNode) -> bool:
        if not p or not q:
            return p == q
        return p.val == q.val and self.helper(p.left, q.right) and self.helper(p.right, q.left)

File: solutions/cpp/0101.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return helper(root, root);
    }

private:
    bool helper(TreeNode* p, TreeNode* q) {
        if (!p || !q) return p == q;
        return p->val == q->val &&
               helper(p->left, q->right) &&
               helper(p->right, q->left);
    }
};